### ISSUE TITLE
Multiple window link does not function in native automation with target = "_other" & disableMultipleWindows

### ISSUE THREAD

**Issue by Eng-raghad** [DevExpress/testcafe#8045]
### What is your Scenario?

If you try to open a link with `target= "_other"` & `disableMultipleWindows: true` using v 3.3.0, it does not function with the default native automation.

 

### What is the Current behavior?

TestCafe can not open the link

### What is the Expected behavior?

Link can be opened within the same window

### What is your public website URL? (or attach your complete example)

https://www.mailinator.com/

### What is your TestCafe test code?

```
 await t.navigateTo("https://www.mailinator.com/")
 await t
  .typeText(this.txt_email_search, your_mailinator_email, { replace: true, paste: true })
  .click(Selector('button').withAttribute('value', 'Search for public inbox for free'))
  .click(Selector('tbody').find('td').withText(your_email_title))
  .click(Selector('#pills-links-tab'))
  .click(Selector('a').withAttribute('href', "https://www.google.com/"))
```

### Steps to Reproduce

1- Go to mailinator.com
2- Enter any name for the email. Ec: testlink. It will generate an open email as testlink@mailinator.com
3- From your personal mail, send email to mailinator email with link
4- In your TestCafe test, open mailinator inbox, click on "Links", Click to open the link
5- Observe that the link element has attribute `target= "_other"`: https://d.pr/i/Cwj237 

### TestCafe version

3.3.0

### Node.js version

v16.13.0

### Command-line arguments

chrome

### Browser name(s) and version(s)

118

### Platform(s) and version(s)

MacOS 13.5.2

**Comment by Eng-raghad** [comment]
Hi @PavelMor25, my apologies. I've shared the details with support@devexpress.com. It's odd that the click action failed to occur on my side, yet the error didn't originate from the click step; instead, it manifested in the subsequent step, despite the absence of the click execution.

**Comment by PavelMor25** [comment]
Hello @Eng-raghad,

We have received your message and reviewed it. Unfortunately, we didn't manage to determine why the issue occurs based on the video.

>What options do we have to provide temporary access to the repo with the failed example outlined in the video?

If you would like the TestCafe team to investigate the issue, please share here an example of your test code without any sensitive information and write detailed instructions on how to reproduce the issue. 

Meanwhile, you can use a workaround with disabling the native automation, while we are researching the issue.

Refer to the following help topic: [how to create minimal working example](https://testcafe.io/documentation/402636/faq/general-info#how-to-create-a-minimal-working-example-when-you-submit-an-issue).

**Comment by Eng-raghad** [comment]
@PavelMor25  Here are the details of the test:


### Test Code: 
URL, customer center url, & contact email are shared via email.
``` 
fixture `sites - courses 1`
    .page(URL);
test
  .meta({ created_on: '02/15/22', created_by: 'Patrick Duff', updated_on: '04/06/23', updated_by: 'Raghad Al Fatal' })
  .meta({ scope: 'sanity', native_automation: 'false', priority: 'high', showstopper: 'no', github_ID: '446' })
  .before(async t => {
    await t.navigateTo(await HelperObjects.get_customer_center_url()) 
  })
  ('user_can_log_into_customer_center_with_magic_link', async t => {

    // Purpose: To verify that a user can use a magic link email to log in to the customer center.

    const contact_email = data.users.magic_link_user.customer_center_email
    const mailinator_url = 'https://www.mailinator.com/'
    const email_title = "My Workspace\'s Site - Magic Link Request"

    await CustomerCenterObjects.request_magic_link(contact_email)

    // Adding this to avoid race condition where it starts searching for the email in Mailinator before the navigation begins  
    await HelperObjects.wait_for_flash_alert()

    await t.navigateTo(mailinator_url)
    await MailinatorPageObjects.search_for_email(contact_email, email_title)
    await MailinatorPageObjects.login_with_magic_link(email_title)

    // Verify that user has access to Customer Center
    await t
      .expect(Selector('h1').withText('All Courses').visible).ok()
  })
```  
  
**customer_center_repo.js:**
```
import { Selector, t } from 'testcafe'

class CustomerCenterObjects {
    constructor() {
      this.txt_field_email = Selector('input').withAttribute('data-testid', 'contact_email')
      this.btn_request_magic_link = Selector('input').withAttribute('value', 'Request magic link')
    }
  
    async request_magic_link(user_email){
  
      //Purpose: Fills in email and clicks on the magic link button
      //Param: user_email - String - uses magic link user email
      
      await t
        .typeText(this.txt_field_email, user_email, { replace: true, paste: true })
        .click(this.btn_request_magic_link)
    }
  }
export default new CustomerCenterObjects()
```

**helpers.js:**
```
import { Selector, t } from 'testcafe'

class HelperObjects {
    constructor() {
      this.flash_alert = Selector('#toaster').child('div').withAttribute('data-controller', 'alert')
    }
  
  
    async wait_for_flash_alert(checks = 30) {
  
      // Purpose: To wait for the flash alert to appear and then disappear
      // Param checks - Int
  
      let count = 0
  
      while (await this.flash_alert.visible == false && count < checks) {
        await t
          .wait(250)
  
        count++
      }
  
      count = 0
      
      while (await this.flash_alert.visible && count < checks) {
        await t
          .wait(250)
  
        count++
      }
    }
  
  async wait_for_element(element, visible_or_exists = 'visible', boolean, count = 10) {
  
      // Purpose: To wait for an element
      // Param: element - testcafe Selector - locator
      // Param: visible_or_exists - String - Check if the element is visible or exists
      // Param: boolean - Bool - true/false
      // Param: count - Int - The amount of loops we will be waiting for
  
      // This is a custom request hook that is waiting for a failed request.
      // await CustomRequestHooks.waitForFailedRequest(5000)
  
      switch(visible_or_exists) {
        case 'visible':
          while (await element.visible == boolean && count > 0) {
            await t
              .wait(250)
              count--
          }
          break
  
        case 'exists':
          while (await element.exists == boolean && count > 0) {
            await t
              .wait(250)
              count--
          }
          break
      }
    }
  
  
  
  }
  export default new HelperObjects()  
  ```
  
  **mailinator_page_repo.js:**
  ```
  import HelperObjects from '../pages/helpers'
import { Selector, t } from 'testcafe'


class MailinatorPageObjects {
    constructor() {
      this.txt_email_search = Selector('#search').withAttribute('placeholder', 'Enter Public Mailinator Inbox')
      this.btn_go_to_email = Selector('button').withAttribute('value', 'Search for public inbox for free')
      this.email_received_time = Selector('tbody').find('td').withText(/(minute|just now)/)
      this.link_links = Selector('#pills-links-tab')
      this.link_login_link = Selector('a').withAttribute('href', /.*(magic_link|reset_password).*/)
  
  
    }
  
   async search_for_email(user_email, email_title) {
  
      // Purpose: To search for an email address within mailinator
      // Param: user_email - string
      // Param: email_title - string
  
      await HelperObjects.wait_for_element(this.txt_email_search, 'visible', false, 100)
  
      await t
        .typeText(this.txt_email_search, user_email, { replace: true, paste: true })
        .click(this.btn_go_to_email)
  
      await HelperObjects.wait_for_element(this.email_received_time.sibling('td')
                                          .withText(email_title),'visible', false, 50)
    }
  
   async login_with_magic_link(email_title){
  
      // Purpose: Navigates to the links tab and clicks the magic link
      // Param: email_title - string
  
      await t
        .click(this.email_received_time.sibling('td').withText(email_title))
        .click(this.link_links)
  
      await HelperObjects.wait_for_element(this.link_login_link, 'visible', false)
  
      await t.click(this.link_login_link)
    }
}  
export default new MailinatorPageObjects()  
```

### Your complete configuration file - testcaferc.js:
```
require('dotenv').config()


// Agnostic for all environments
let skipped_tests = [

    // Located in affiliate_center_page_tests.js
    'user_can_add_new_affiliate_campaign', // 08-09-23

    // Located in all_contacts_page_tests.js
    'user_can_modify_contacts_table_columns', // 12-19-22
    'user_can_edit_contact_from_index', // 05-01-23
    'user_can_edit_contact_from_full_profile_page', // 01-09-23
    
    // Located in all_products_page_tests.js
    'user_can_create_a_subscription_product', // 09-27-22
    'user_can_set_a_price_for_physical_products', // 09-27-22
    'user_can_create_a_product_with_a_one_time_price_of_free', // 08-28-23

    // Located in product_variant_tests.js
    'user_can_add_one_time_price_to_product_variant', // 10/19/23
    'user_can_add_payment_plan_price_with_paid_trial_to_product_variant', // 10/19/23
    'user_can_add_subscription_price_with_free_trial_to_product_variant', //10/19/23

    // Located in billing_page_tests.js
    'user_can_navigate_to_billing_page', // 08-22-22

    // Located in blog_page_tests.js
    'user_is_able_to_change_blog_template', //04-04-23

    // Located in courses_page_tests_1.js
    'manually_added_user_can_navigate_through_course', // 05-09-23
    'courses_back_button_navigates_to_previous_page', // 07-05-23

    // Located in courses_page_tests_2.js
    'user_can_bulk_publish_lessons_in_course', // 10/20/23

    // Located in courses_page_tests_3.js
    'user_can_override_module_with_a_template_in_course_settings', // 06-26-23
    'user_can_override_lesson_with_a_template_in_course_settings', // 06-26-23
    'user_can_override_lesson_with_a_template_in_lesson_settings', // 06-26-23

    // Located in dashboard_page_tests.js
    'user_can_navigate_to_forms_page_from_dashboard', // 11-22-22
    'user_can_navigate_to_payouts_page_from_dashboard', // 11-22-22

    // Located in funnels_page_tests.js
    'user_can_clone_a_funnel',
    'user_can_navigate_through_funnel',
    'user_can_restore_an_archived_funnel', // 03-16-23
    'page_views_and_unique_views_values_are_correct', // 08-09-23

    // Located in funnels_workflow_builder_page_tests.js
    'funnel_page_seo_settings_are_rendered', // 07-20-23
    
    // Located in marketing/broadcasts_page_tests.js
    'user_can_create_new_broadcast', // 06-20-22

    // Located in marketing/templates_page_tests.js
    'user_can_create_broadcast_template', // 06-13-22

    // Located in orders/order_page_tests.js
    'user_is_able_to_set_delayed_status_on_order', // 09-27-22
    'user_is_able_to_set_processing_status_on_order', // 09-27-22
    'user_is_able_to_set_packed_status_on_order', // 09-27-22
    'user_is_able_to_set_ready_status_on_order', // 09-27-22
    'user_is_able_to_set_delivered_status_on_order', // 09-27-22
    'user_is_able_to_set_shipped_status_on_order', // 09-27-22
    'user_is_able_to_set_backorder_status_on_order', // 09-27-22
    'user_is_able_to_set_pending_status_on_order', // 09-27-22
    'user_is_able_to_set_lost_status_on_order', // 09-27-22
    'user_can_add_multiple_fulfillments_to_order', // 09-27-22
    'user_can_partially_fulfill_order', // 08-11-22

    // Located in order_funnel_page_tests.js
    'taxes_are_reflected_in_order_checkout_and_confirmation',

    // Located in pages_page_tests.js
    'user_can_pin_a_page_from_edit_page', //07-29-22
    'user_can_change_page_style', //10/21/23
    'user_can_edit_page_information', //10/21/23

    // Located in payments_ai_tests.js
    'user_can_navigate_to_payments_ai_dashboard', //03-08-23

    // Located in pipelines_page_tests.js
    'user_can_navigate_to_pipelines_page', // 09-14-22

    // Located in reporting_page_tests.js
    'opt-ins_by_landing_page_report_displays', // 06-21-22
    'sales_by_product_report_displays', // 01-18-23

    // Located in sites_overview_page_tests.js
    'user_can_add_header_to_site', // 07-06-22
    'user_can_change_site_style', // 10-31-22
    'user_can_edit_site_name', // 10-04-22
    'user_can_make_site_password_protected', // 10-03-22

    // Located in themes_page_tests.js
    'user_can_edit_theme_name', // 08-08-22

    // Located in url_redirects_page_tests.js
    'user_can_delete_a_url_redirect', // 11-30-22
    'user_is_able_to_edit_url_redirect', // 11-30-22

    // Located in editor_page_tests.js
    'clicking_button_hides_shows_elements' // 03-16-23
]

// List of tests that open new window and thus needs native-automation to be disabled
let multiple_window_tests = [
    'user_can_log_into_customer_center_with_magic_link',
    'user_is_able_to_add_a_section_column_row_and_element_to_editor',
    'saving_is_successful_when_adding_a_video',
    'user_is_able_to_save_when_leaving_video_embed_blank',
    'user_can_change_page_style',
    'user_can_create_a_new_account_through_funnel',
    'user_can_create_a_new_account_through_uncached_funnel',
    'page_views_and_unique_views_values_are_correct'
]

// Cleanup scripts that need to be skipped, but can't be in the skipped tests
let clean_up_tests = [
    // Located in clean_up_scripts.js
    'clean_up_copies_of_funnels',
    'clean_up_copies_of_funnel_pages',
    'clean_up_copies_of_pages', 
    'clean_up_products',
    'clean_up_site_pages',
    'clean_up_workflows',
    'dismiss_notifications',
    'clean_up_collaborators',
]

// Production specific scripts needing to be skipped for various reasons
let production_skipped_tests = [
    'user_can_navigate_to_affiliate_center_page', // Disabling because the link is disabled in production as of 12/07/22
    'user_can_navigate_to_integrations_page', // Disabling because the link is disabled in production as of 01/23/23
    'user_can_navigate_to_order_tags_page', // Link disabled through Launch Darkly as of 08/29/22
    'user_can_navigate_to_payments_ai_dashboard', // Production PAI account not set up as of 08/29/22
    'user_is_able_to_navigate_to_the_inbox_page', // Inbox link is not activated in production as of 08/31/22
    'user_can_navigate_to_sequences_page', // Disabling because it's behind a feature gate in produciton as of 10/17/22
    'user_can_navigate_to_countdown_events_page', // Disabling because the link is disabled in production as of 11/28/22
    'user_can_navigate_to_community_page', // Disabling because the link doesn't display in production as of 12/21/22
    'user_can_navigate_to_the_countdown_timers_page', // Disabling because the link is disabled in production as of 01/03/23
]

// Scripts that need to run with a single concurrency
let isolated_tests = [
    // Located in sites_overview_page_tests.js
    'user_can_make_site_public',
    'user_can_edit_site_name',
    'user_can_make_site_password_protected',
    'user_can_make_site_private',
    'user_can_edit_site_domain',
    'changing_site_domain_updates_funnel_urls',
    'changing_site_domain_updates_blog_urls',
    'changing_site_domain_updates_course_urls',
    'user_can_add_header_to_site',

    // Located in account_details_page_tests.js
    'user_can_edit_account_name',
    'user_can_edit_account_password',
    'user_can_change_account_language',

    // Located in general_page_tests.js
    'user_can_edit_workspace_details',

    // Located in blog_page_tests.js
    'user_can_edit_blog_settings',

    // Located in themes_page_tests.js
    'user_can_promote_theme_to_live_from_index_page',
    'user_can_promote_theme_to_live_from_overview_page',

    // Located in the workflow_builder_page_tests.js
    'broadcast_room_category_pulls_back_expected_pages',
    'downsell_category_pulls_back_expected_pages',
    'landers_category_pulls_back_expected_pages',
    'my_funnel_pages_category_pulls_back_expected_pages',
    'opt_in_category_pulls_back_expected_pages',
    'order_category_pulls_back_expected_pages',
    'order_confirmation_category_pulls_back_expected_pages',
    'sales_category_pulls_back_expected_pages',
    'terms_of_service_category_pulls_back_expected_pages',
    'thank_you_category_pulls_back_expected_pages',
    'upsell_category_pulls_back_expected_pages',
    'webinar_confirmation_category_pulls_back_expected_pages',
    'webinar_registration_category_pulls_back_expected_pages'
]

// Filter that joins the skipped scripts with the production skipped tests with cleanup and multiple windows tests
let production_with_multi_windows = skipped_tests.concat(production_skipped_tests).concat(clean_up_tests).concat(multiple_window_tests)

// Filter that joins the skipped scripts with the production skipped with cleanup tests
let production_filter = skipped_tests.concat(production_skipped_tests).concat(clean_up_tests)

// Filter that joins ALL filtered scripts so that they are not run for a full suite
let full_filter = skipped_tests.concat(isolated_tests).concat(clean_up_tests).concat(multiple_window_tests)

// Filter that joins ALL skipped tests and tests that deal with multiple windows 
let skipped_and_multi_windows = skipped_tests.concat(multiple_window_tests)

let build_filter = () => {

    // Purpose: To set the suite that we want to run. (isolated, regression, sanity)

    var filter = {}
    
    switch(process.env.ENV) {
        case 'production':
            if((process.env.SCOPE) && (process.env.NATIVE_AUTOMATION)) {
                filter = {
                    testGrep: `^(?!.*(${production_filter.join('|')})).*$`, 
                    testMeta: {
                        scope: process.env.SCOPE,
                        native_automation: process.env.NATIVE_AUTOMATION
                    }
                }
            }
            else if(process.env.SCOPE) {
                filter = {
                    testGrep: `^(?!.*(${production_with_multi_windows.join('|')})).*$`,
                    testMeta: {
                        scope: process.env.SCOPE
                    }
                }
            }
            else if(process.env.FREQUENCY) {
                filter = {
                    testGrep: `^(?!.*(${production_filter.join('|')})).*$`,
                    testMeta: {
                        frequency: process.env.FREQUENCY
                    }
                }
            }
            break
        case 'staging':
            if((process.env.SCOPE) && (process.env.NATIVE_AUTOMATION)) {
                filter = {
                    testGrep: `^(?!.*(${skipped_tests.join('|')})).*$`, 
                    testMeta: {
                        scope: process.env.SCOPE,
                        native_automation: process.env.NATIVE_AUTOMATION
                    }
                }
            }
            else if(process.env.SCOPE) {
                filter = {
                    testGrep: `^(?!.*(${skipped_and_multi_windows.join('|')})).*$`, 
                    testMeta: {
                        scope: process.env.SCOPE
                    }
                }
            }
            else if(process.env.NATIVE_AUTOMATION) {
                filter = {
                    testGrep: `^(?!.*(${skipped_tests.join('|')})).*$`,
                    testMeta: {
                        native_automation: process.env.NATIVE_AUTOMATION
                    }
                }
            }
            else if(process.env.FREQUENCY) {
                filter = {
                    testGrep: `^(?!.*(${skipped_tests.join('|')})).*$`,
                    testMeta: {
                        frequency: process.env.FREQUENCY
                    }
                }
            }
            else {
                filter = {
                    testGrep: `^(?!.*(${full_filter.join('|')})).*$`
                }
            }
            break
    }

    return filter
}

let determine_concurrency = () => {

    // Purpose: To set concurrency dependant on the scope that is passed
    
    let concurrency = 5
    if (process.env.SCOPE == 'isolated') {
        concurrency = 1
    } 
    return concurrency
}

module.exports = {
    assertionTimeout: 5000,
    browsers: [
        "chrome:headless --window-size=1920,1159"
    ],
    concurrency: determine_concurrency(),
    cache: true,
    clientScripts: [
        "scripts/click_expand_all_for_courses_details_page.js",
        "scripts/hide_userflow_icon.js"
    ],
    disableMultipleWindows: true,
    filter: build_filter(),
    hostname: "localhost",
    pageLoadTimeout: 30000,
    quarantineMode: {
        successThreshold: 1,
        attemptLimit: 2
    },
    reporter: [
        {
            name: "spec",
            output: "artifacts/reports/spec_results"
        },
        {
            name: "xunit",
            output: "artifacts/reports/xunit_results.xml"
        }, 
        {
            name: "testrail"
        }
    ],
    runExecutionTimeout: 4500000,
    screenshots: {
        path: "artifacts/screenshots",
        pathPattern: "${TEST}_${DATE}_${TIME}.png",
        takeOnFails: true,
        thumbnails: false
    },
    selectorTimeout: 5000,
    skipJsErrors: true,
    skipUncaughtErrors: true,
    src: "tests/**",
    testExecutionTimeout: 600000
}
```

### TestCafe version: 3.4.0

### Your complete test report:
```
 Running tests in:
 - Chrome 119.0.0.0 / Ventura 13

 sites - courses 1
 ✖ user_can_log_into_customer_center_with_magic_link (screenshots: )

   1) AssertionError: expected false to be truthy

      Browser: Chrome 119.0.0.0 / Ventura 13
      Screenshot: 

         306 |    await MailinatorPageObjects.search_for_email(contact_email, email_title)
         307 |    await MailinatorPageObjects.login_with_magic_link(email_title)
         308 |
         309 |    // Verify that user has access to Customer Center
         310 |    await t
       > 311 |      .expect(Selector('h1').withText('All Courses').visible).ok()
         312 |  })
         313 |
         314 |test
         315 |  .meta({ created_on: '2/17/22', created_by: 'Patrick Duff', updated_on: '03/02/23', updated_by: 'Justin Grabenbauer' })
         316 |  .meta({ scope: 'regression', priority: 'low', showstopper: 'no', CID: 'C45' })

         at <anonymous> (/Users/raghadalfatal/Desktop/cf2testcafe/cf2-test-cafe-poc/tests/sites/courses/courses_page_tests_1.js:311:63)
         at asyncGeneratorStep (/Users/raghadalfatal/Desktop/cf2testcafe/cf2-test-cafe-poc/tests/sites/courses/courses_page_tests_1.js:21:94)
         at _next (/Users/raghadalfatal/Desktop/cf2testcafe/cf2-test-cafe-poc/tests/sites/courses/courses_page_tests_1.js:21:94)

 1/1 failed (1m 01s)

 Warnings (2):
 --
  The "browsers", "concurrency", and "filter" options from the configuration file will be ignored.
```

### Steps to Reproduce:
Run the script:
`ENV=staging testcafe chrome -t 'user_can_log_into_customer_center_with_magic_link' -c1`

### Node.js version
v16.13.0

### Browser name(s) and version(s)
119.0.0

### Platform(s) and version(s)
MacOS Ventura 13.5.2

**Comment by Eng-raghad** [comment]
Hello @aleks-pro, here's the code sample. Note that some data is shared with @PavelMor25 via email. Please let me know if anything else is needed.
[multiple_windows on native_automation.zip](https://github.com/DevExpress/testcafe/files/13481687/multiple_windows.on.native_automation.zip)

**Comment by Eng-raghad** [comment]
@aleks-pro Checking to see if there's any update on this.

**Comment by Bayheck** [comment]
Hello,

We are still working on this issue and we don’t have any updates yet.
However, we have released TestCafe v3.5.0, which offers an experimental solution for [multi-window tests in native automation](https://testcafe.io/404732/release-notes/framework/2023-12-28-testcafe-v3-5-0-released#experimental-run-multi-window-tests-with-native-automation). You can use the [--experimental-multiple-windows](https://testcafe.io/documentation/402639/reference/command-line-interface#--experimental-multiple-windows) CLI flag to activate this mode. Please let us know if you encounter any issues.
